packages/alfa-rules/src/sia-r113/rule.ts

```diff
@@ -50,96 +69,120 @@ export default Rule.Atomic.of<Page, Element>({
  * @public
  */
 export namespace Outcomes {
-  export const HasSufficientSizeOrSpacing = (name: string, box: Rectangle) =>
+  export const IsUserAgentControlled = (name: string, box: Rectangle) =>
     Ok.of(
-      WithBoundingBox.of("Target has sufficient size or spacing", name, box),
+      WithBoundingBox.of(
+        "Target is user agent controlled",
+        name,
+        box,
+        Either.left({ ua: true }),
+        [],
+      ),
     );
 
-  export const HasInsufficientSizeAndSpacing = (name: string, box: Rectangle) =>
-    Err.of(
-      WithBoundingBox.of("Target has insufficient size and spacing", name, box),
+  export const HasSufficientSize = (name: string, box: Rectangle) =>
+    Ok.of(
+      WithBoundingBox.of(
+        "Target has sufficient size",
+        name,
+        box,
+        Either.right({ size: true, spacing: true }),
+        [],
+      ),
     );
 
-  export const IsUserAgentControlled = (name: string, box: Rectangle) =>
-    Ok.of(WithBoundingBox.of("Target is user agent controlled", name, box));
+  export const HasSufficientSpacing = (name: string, box: Rectangle) =>
+    Ok.of(
+      WithBoundingBox.of(
+        "Target has sufficient spacing",
+        name,
+        box,
+        Either.right({ size: false, spacing: true }),
+        [],
+      ),
+    );
+
+  export const HasInsufficientSizeAndSpacing = (
+    name: string,
+    box: Rectangle,
+    tooCloseNeighbors: Iterable<Element>,
+  ) =>
+    Err.of(
+      WithBoundingBox.of(
+        "Target has insufficient size and spacing",
+        name,
+        box,
+        Either.right({ size: false, spacing: false }),
+        tooCloseNeighbors,
+      ),
+    );
 }
 
-function hasSufficientSizeOrSpacing(
+function hasSufficientSpacing(
   document: Document,
   device: Device,
-): Predicate<Element> {
-  return or(
-    hasSufficientSize(24, device),
-    hasSufficientSpacing(document, device),
-  );
+): Predicate<Element, [Array<Element>]> {
+  return (target, tooCloseNeighbors) => {
+    for (const underspacedOther of findElementsWithInsufficientSpacingToTarget(
+      document,
+      device,
+      target,
+    )) {
+      tooCloseNeighbors.push(underspacedOther);
+    }
+    return tooCloseNeighbors.length === 0;
+  };
 }
 
-const spacingCache = Cache.empty<
+const undersizedCache = Cache.empty<
   Document,
-  Cache<Device, Cache<Element, boolean>>
+  Cache<Device, Sequence<Element>>
 >();
 
 /**
  * @remarks
- * Spacing is calculated by
- * 1. drawing a 24px diameter circle around the center of the bounding box of the target,
- * 2. checking if the circle intersects with the bounding box of any other target, or
- * 3. if the circle intersects with the 24px diameter circle of another undersized target.
+ * This generator functions yields all elements that have insufficient spacing to the target.
```

Comment:
**Chore:** move comment title before the `@remarks` tag.
(can't make suggestion on that one, not sure why 🤔)